src/agents/workflows/subgraphs/step1_preprocessing/vision_llm.py

```
@@ -0,0 +1,165 @@
+            ocr_list.append({
+                "page": idx,
+                "ocr_text": f"Mock OCR text for {img_path.name}",
+                "blocks": [{"type": "latex", "text": "a/b", "latex": "\\frac{a}{b}", "bbox": [0,0,0,0]}]
+            })
+        return {"ocr": ocr_list, "image_caption": []}
+
+class GeminiVisionProvider(VisionProvider):
+    def __init__(self, model_name: str = "models/gemini-2.5-flash"):
+        success, lib = check_genai()
+        if not success:
+            raise RuntimeError("Gemini 라이브러리가 없습니다.")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)